src/easylink/step.py

```diff
@@ -51,60 +56,120 @@ def validate_step(
         pass
 
     @abstractmethod
-    def update_implementation_graph(self, graph: nx.MultiDiGraph) -> None:
+    def get_implementation_graph(self) -> ImplementationGraph:
         """Resolve the graph composed of Steps into a graph composed of Implementations."""
         pass
 
-    def set_parent_step(self, step):
+    @abstractmethod
+    def get_implementation_edges(self, edge: Edge) -> List[Edge]:
+        """Propagate edges of StepGraph to ImplementationGraph."""
+        pass
+
+    def set_parent_step(self, step) -> None:
         self.parent_step = step
 
-    def configure_step(self, step_config: LayeredConfigTree) -> None:
+    def set_step_config(self, parent_config: LayeredConfigTree) -> None:
+        self._config = parent_config[self.name]
+
+    def configure_step(
+        self, step_config: LayeredConfigTree, input_data_config: LayeredConfigTree
+    ) -> None:
         self.set_step_config(step_config)
 
 
 class IOStep(Step):
     """"""
 
     @property
-    def pipeline_graph_node_name(self):
+    def implementation_graph_node_name(self):
         return "pipeline_graph_" + self.name
```

Comment:
The mismatch between the name and value of this property is confusing.

Reply:
I shouldn't actually need this prefix anyway anymore